bluebottle/accounting/utils.py

```diff
@@ -0,0 +1,293 @@
+from django.db.models import Sum, Count, Q
+from django.utils import timezone
+from datetime import datetime
+
+from .models import BankTransaction, RemoteDocdataPayment, RemoteDocdataPayout, BankTransactionCategory, BankAccount
+from decimal import Decimal
+
+
+def add_up(first, second):
+    """
+    Addition of two objects.
+
+    Distinguish addition of dictionaries, list and numbers.
+
+    return None when both are None
+    if one of the two is None, return the other one which is not None
+
+    if both are mydicts, add them up with the custom mydict.__add__
+    if both are mylists, add them up with the .add method
+
+    if both are convertable to a Decimal, return the addition of the numbers
+
+    in other cases, return None
+    """
+    if not first:
+        return second
+    elif not second:
+        return first
+    else:
+        if isinstance(first, mydict) and isinstance(second, mydict):
+            return first + second
+        elif isinstance(first, mylist) and isinstance(second, mylist):
+            return first.add(second)
+        else:
+            try:
+                return Decimal(str(first)) + Decimal(str(second))
+            except:
+                return None
+
+
+class mydict(dict):
+
```

Comment:
I have forwarded it to Jos who wrote this initially, I'm in a meeting all
day.
On Jan 6, 2016 09:57, "Loek van Gent" <notifications@github.com> wrote:

> In bluebottle/accounting/utils.py
> <https://github.com/onepercentclub/bluebottle/pull/1794#discussion_r48937797>
> :
>
> > +    elif not second:
> > +        return first
> > +    else:
> > +        if isinstance(first, mydict) and isinstance(second, mydict):
> > +            return first + second
> > +        elif isinstance(first, mylist) and isinstance(second, mylist):
> > +            return first.add(second)
> > +        else:
> > +            try:
> > +                return Decimal(str(first)) + Decimal(str(second))
> > +            except:
> > +                return None
> > +
> > +
> > +class mydict(dict):
> > +
>
> @sergei-maertens <https://github.com/sergei-maertens> could you shine
> some light on this?
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/onepercentclub/bluebottle/pull/1794/files#r48937797>.
>